### PR TITLE
fix(cactus-example-discounted-asset-trade): enable tsconfig strict flag and fix all the warnings

### PR DESCRIPTION
**Pull request by rwAT17**
- Fixed all warnings when strict flag is enabled

Closes: #2145

Signed-off-by: Tomasz Awramski <tomasz.awramski@fujitsu.com>

### CODE REVIEW

**Comment by rwAT17** [comment]
@izuru please review

**Comment by takeutak** [comment]
@rwAT17 Thanks for contribution and I'm very sorry to wait you for a while! I'm asking @izuru0 to review this PR again. He will review it again next Monday, so please wait for a minute?

File: examples/cactus-example-discounted-asset-trade/balance-management.ts [hunk 1]
```diff
@@ -41,7 +44,7 @@ export class BalanceManagement {
       // for Neo
       const contract = {}; // NOTE: Since contract does not need to be specified, specify an empty object.
       const method = { type: "web3Eth", command: "getBalance" };
-      const template = "default";
+      // const template = "default";
```
Comment (izuru0) [inline examples/cactus-example-discounted-asset-trade/balance-management.ts:47]: hi
can you remove unused codes, including already commented out lines such as `// const method = "default";` below?
thanks

Comment (rwAT17) [reply examples/cactus-example-discounted-asset-trade/balance-management.ts]: Sorry you had to wait, but i somehow missed notification on github about your comment. Did what you asked for.

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM

**Review by izuru0 — APPROVED** [review]
LGTM

**Pull request merged**